src/locales/fa/location/direction.ts

```
@@ -1,10 +1,6 @@
-export default [
-  'شمال',
-  'شرق',
-  'جنوب',
-  'غرب',
-  'شمال شرق',
-  'شمال غرب',
-  'جنوب شرق',
-  'جنوب غرب',
-];
+export default {
+  cardinal: ['شمال', 'شرق', 'جنوب', 'غرب'],
+  cardinal_abbr: [],
```

Note this and similar changes are a behavior change, previously
fakerFA.location.direction({abbreviated:true}) would fall back to English and return say NE

Now it throws
[Error]: Cannot get value from empty dataset.

I think thats OK but it should probably be considered a breaking change for this reason

---

Alternatively if there is no data for cardinal_abbr etc then the abbreviated option should be ignored and the full name returned.

---

I think we should go for breaking then.

---

Good catch. I left them like this instead of making the subset undefined  or even null because I think that this data has been forgotten to be implemented. Those values are part of the legacy package. I was not able to figure out if abbreviated content is not applicable in those locales. By throwing people with knowledge about these locales can tell us the expected behavior:
- abbreviated is same as wide context
- abbreviated context does not exist in those locales
- implement the correct abbreviated data

---

agree, though it would be better if it threw the helpful "The locale data for.... are missing in this locale" error rather than the mysterious "Cannot get value from empty dataset" error

---

In that case, I could wrap the current type in the `LocaleEntry` type to allow `undefined` (and `null`) values. `undefined` should throw the error you suggest. I would be fine with that.

---

Note since N, E, S, W are included in CLDR (for use in geographical coordinates) its fairly easy to look up what the equivalent of "N", "E", "S", "W" is in most languages e.g.:

https://www.unicode.org/cldr/charts/42/by_type/units.coordinates.html#2e784791aa57e39c

---

In examples like fa or ja where there is no abbreviated version, what should we do? In that case it seems more sensible to repeat the full version in the abbreviated field.

---

Since data has now been added for the missing locales I would no longer consider this breaking.